### PR TITLE
minor refactor GenerativeAgentMemory

### PR DESCRIPTION
**Pull request by mbchang**
# minor refactor of GenerativeAgentMemory

- refactor `format_memories_detail` to be more reusable
- modified prompts for getting topics for reflection and for generating insights
- update `characters.ipynb` to reflect changes

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

@vowelparrot
@hwchase17
@dev2049

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
@vowelparrot can you review?

File: docs/use_cases/agent_simulations/characters.ipynb
Comment (vowelparrot) [inline docs/use_cases/agent_simulations/characters.ipynb:1]: Is it intentional that the 'eve overhears her colleagues' is commented out?

**Comment by vowelparrot** [comment]
@mbchang apart from the notebook commented out line, the rest lgtm

File: docs/use_cases/agent_simulations/characters.ipynb
Comment (mbchang) [reply docs/use_cases/agent_simulations/characters.ipynb:1]: Oh oops, I meant to remove it. When I was running the code it seemed like that sentence was both out-of-place and unnecessary (given the other observations). I have deleted the comment.

**Review by vowelparrot — APPROVED** [review]

**Pull request merged**